lib/jwt/jwt.go

```diff
@@ -145,12 +145,13 @@ func (p *SignParams) Check() error {
 }
 
 // sign will return a signed JWT with the passed in claims embedded within.
-func (k *Key) sign(claims any) (string, error) {
-	return k.signAny(claims)
+// `opts`, when not nil, specifies additional signing options, such as additional JWT headers.
+func (k *Key) sign(claims any, opts *jose.SignerOptions) (string, error) {
+	return k.signAny(claims, opts)
 }
 
 // signAny will return a signed JWT with the passed in claims embedded within; unlike sign it allows more flexibility in the claim data.
-func (k *Key) signAny(claims any) (string, error) {
+func (k *Key) signAny(claims any, opts *jose.SignerOptions) (string, error) {
```

Comment:
nit: Instead of passing `nil` in all places where options don't need to be set, you could let pass `WithHeader(Jose.HeaderKey("kid"), ")` as `...options`, so if not needed it can be skipped.

Reply:
You can totally do that. You just need to write a bit of boiler plate to create a bunch (in our case 1) apply function that you can then apply to SignerOptions.